### ISSUE TITLE
Error with large mechanical to electrical systems.

### ISSUE THREAD

**Issue by GrimmDemon** [jrddunbr/ElectricalAge#163]
I was constructing a power station for my arena, or more exactly, I was modifying my existing power station to run more efficiently without exploding during startup.
At the very end, as I was about to begin the startup sequence, I encountered a NaNMJ, NanMrad/s issue. As soon as I saw this, my first expression was: WTH?. Just thought I'd report this bug

**Comment by GrimmDemon** [comment]
![2019-07-10_17 18 07](https://user-images.githubusercontent.com/52752989/60982739-fe36a180-a338-11e9-90e2-ae45f8a59d0a.png)
Dunno how to upload the image. Hopefully this one came through.

**Comment by jrddunbr** [comment]
Huh. Tasty. What version are you using?

**Comment by GrimmDemon** [comment]
I'm using the latest version, released a few days ago. 1.6.2, I believe.

**Comment by jrddunbr** [comment]
Cool, thanks.

**Comment by GrimmDemon** [comment]
Sorry. I saw my error, I corrected it in an edit

**Comment by jrddunbr** [comment]
The current solution is to remove any NaN'd objects. This can include the entire shaft, generators, turbines, any cables with NaNM degrees C, etc. If you see a NaN, it's likely to propagate across the entire circuit network. Yeeah, the bug sucks. 1.15.3 doesn't have it to my knowledge.

**Comment by GrimmDemon** [comment]
Just a few days ago I was using 1.12* (can't remember). The structure I am upgrading has so many cables that I had to improve to the high current cables so that I could transmit 51200 volts (awesome cables by the way) that I took me a few hours to find them and do so. I updated to 1.16.2 from 1.16.1.1 earlier today since I was missing my key blinded wrench (so useful). I've encountered this issue with 1.16.1.1, but I thought that it was already reported.

**Comment by GrimmDemon** [comment]
Rebuilding my system wasn't hard as I had the layout already. Just removed everything and started, slowly, from scratch, monitoring for any NaN comments.

**Comment by GrimmDemon** [comment]
Now that you mentioned it. Propagate across the ENTIRE network? Aw gees, my miles of cables...

**Comment by jrddunbr** [comment]
Ah. Bringing voltages up to 51kV on the new current cables may be causing the problem (I've seen it happen in preliminary tests). Also, make sure that the `cableResistanceMultipler` in the config is up around 10 billion or so. Sounds big, but you're moving a decimal point over to reduce problems. It should be the default.

NaN's can't cross (from my experience) over isolated DC/DC converters.

**Comment by jrddunbr** [comment]
Calling 1.16.x stable would be highly misleading btw.

**Comment by GrimmDemon** [comment]
OK, so my cables were saved by the transformers directly after the generators.
seems like the NaN issue remained with the shaft system itself.
I noticed once that you said that you were going to put a voltage limit on the current orientated cables. I might have a few explosion if the voltages were below 64000 volts 😂

**Comment by GrimmDemon** [comment]
Err.. BTW, where might I find the cable resistance multiplier. I've managed to take the high current cables to 1.02 million volts without exploding, but I didn't deem that much voltage necessary

**Comment by GrimmDemon** [comment]
I'm not much of a coder, I'm although very good at electrical engineering and zml coding, didn't study much Java.
My main profession is computer infrastructure

**Comment by jrddunbr** [comment]
`cableResistanceMultipler` is in the Eln.cfg file under the `debug` section.

I'm not sure what the voltage will bet yet. Probably a top of 100kV is what I've been thinking most recently.

**Comment by GrimmDemon** [comment]
Is there any special software that I need to open the .jar file? My Java editor isn't opening it, or am I missing something. Currently I'm limited to winrar for opening the .jar file and notepad++ for .xml/.html coding.

**Comment by GrimmDemon** [comment]
Found it. Now I feel like an idiot

**Comment by GrimmDemon** [comment]
I take it this is what I'm supposed to be looking for. 
![Screenshot (4)](https://user-images.githubusercontent.com/52752989/60988434-e44e8c00-a343-11e9-8251-9e41f13eec3c.png)

**Comment by GrimmDemon** [comment]
If it's possible, can I also reduce the noise generated by the generators, they are quite loud?

**Comment by GrimmDemon** [comment]
Thanks for the help. Changing the cable resistance multiplier helped prevent explosions on some of my other electrical systems that I'm trying to port over to a more efficient cable transmission. I am having issues with my wireless network for my light controls, but that seems to only be when my world loads.

**Comment by GrimmDemon** [comment]
Was I supposed to close the issue, or leave it open? I accidentally hit "close and comment", dunno if that changed anything important…

**Comment by jrddunbr** [comment]
Nah, that's fine, you can close it.

If you don't want any noise from generators, you can set the `maxSoundDistance` to zero. That will effectively disable those sounds.

**Comment by GrimmDemon** [comment]
I noticed something funny. Once I set the cable Resistance Multiplier, I didn't get that NaN issue from any of my shaft systems, or maybe I just had a strike of luck, could be that the bug doesn't occur all the time, but I'll keep an eye on it. Also, I thought that changing the Cable Resistance would also solve my one wind farm blowing up, with the voltages being very unstable. It seems like all I had to do was add a constant power using point, such as a light bulb or in my case, a resistor that has (roughly) the same resistance as a 200V LED bulb, otherwise, the cables will all explode when being joined with the rest of the network still using voltage cables.
